### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by breaker05** [CircleCI-Public/aws-cli-orb#46]
Please add support when using the "windows/default" executor. I'm currently use the bash shell as the powershell has some issue too.

**Comment by KyleTryon** [comment]
@breaker05 For your use-case, would you mind the AWS CLI would be installed to Bash on windows, rather than a native windows installation?

**Comment by breaker05** [comment]
I was able to set my executor `shell: bash.exe` and then install the AWS CLI manually and then set the profile variables. Only downside to the _pip install_ is that it will only CLI version 1 so more work would be needed to install the 2.0 CLI package without using pip.

`

      - run:
          name: Install AWS CLI
          command: pip install awscli --upgrade    
       
      - run:
          name: Configure AWS Access Key ID
          command: |
            aws configure set aws_access_key_id \
            $AWS_ACCESS_KEY_ID \
            --profile default
      - run:
          name: Configure AWS Secret Access Key
          command: |            
            aws configure set aws_secret_access_key \
            $AWS_SECRET_ACCESS_KEY \
            --profile default

      - run:
          name: Configure AWS default region
          command: |              
              aws configure set region $AWS_DEFAULT_REGION \
              --profile default
                  
      - run:
          name: Verify AWS config
          command: |
            echo $PATH
            aws --version             
            aws configure list --profile default
`

**Comment by gmemstr** [comment]
Does the `install` command from this orb work under bash? Or is there something specifically incompatible with our implementation and the bash version within Windows? @breaker05

**Comment by kieran-osgood** [comment]
To add to this - I've just tried using the orb on:
 ```          
executor: 
    name: win/default
    shell: bash.exe
```
Noticed the 'latest' was always skipping, and even tried to set the version override found it skipped over it still:

## Step: Install AWS CLI - 2.1.29
```
#!bash.exe
if [[ $EUID == 0 ]]; then export SUDO=""; else export SUDO="sudo"; fi

if [ ! "$(which aws)" ] || [ "$PARAM_AWS_CLI_OVERRIDE" = 1 ]; then
    # setup
    if [ ! "$PARAM_AWS_CLI_VERSION" = "latest" ]; then export AWS_CLI_VER_STRING="-$PARAM_AWS_CLI_VERSION"; fi

    # Uninstall existing AWS CLI if override is enabled.
    if [ "$PARAM_AWS_CLI_OVERRIDE" = 1 ]; then
        AWS_CLI_PATH=$(which aws)
        if [ -n "$AWS_CLI_PATH" ]; then
            EXISTING_AWS_VERSION=$(aws --version)
            echo "Uninstalling ${EXISTING_AWS_VERSION}"
            # shellcheck disable=SC2012
            if [ -L "$AWS_CLI_PATH" ]; then
                AWS_SYMLINK_PATH=$(ls -l "$AWS_CLI_PATH" | sed -e 's/.* -> //')
            fi
            $SUDO rm -rf "$AWS_CLI_PATH" "$AWS_SYMLINK_PATH" "$HOME/.aws/" "/usr/local/bin/aws" "/usr/local/bin/aws_completer" "/usr/local/aws-cli"
        else
            echo "No AWS install found"
        fi
    fi

    echo "Installing AWS CLI v2"
    cd /tmp || exit
    # Platform check
    if uname -a | grep "Darwin"; then
        export SYS_ENV_PLATFORM=macos
    elif uname -a | grep "x86_64 GNU/Linux"; then
        export SYS_ENV_PLATFORM=linux_x86
    elif uname -a | grep "aarch64 GNU/Linux"; then
        export SYS_ENV_PLATFORM=linux_arm
    else
        echo "This platform appears to be unsupported."
        uname -a
        exit 1
    fi
    echo "Platform $SYS_ENV_PLATFORM"
    # Install per platform
    case $SYS_ENV_PLATFORM in
    linux_x86)
        curl -sSL "https://awscli.amazonaws.com/awscli-exe-linux-x86_64${AWS_CLI_VER_STRING}.zip" -o "awscliv2.zip"
        unzip -q -o awscliv2.zip
        $SUDO ./aws/install
        rm awscliv2.zip
        ;;
    macos)
        curl -sSL "https://awscli.amazonaws.com/AWSCLIV2${AWS_CLI_VER_STRING}.pkg" -o "AWSCLIV2.pkg"
        $SUDO installer -pkg AWSCLIV2.pkg -target /
        rm AWSCLIV2.pkg
        ;;
    linux_arm)
        curl -sSL "https://awscli.amazonaws.com/awscli-exe-linux-aarch64${AWS_CLI_VER_STRING}.zip" -o "awscliv2.zip"
        unzip -q -o awscliv2.zip
        $SUDO ./aws/install
        rm awscliv2.zip
        ;;
    *)
        echo "This orb does not currently support your platform. If you believe it should, please consider opening an issue on the GitHub repository:"
        echo "https://github.com/CircleCI-Public/aws-cli-orb/issues/new"
        exit 1
    ;;
    esac
    # Toggle AWS Pager
    if [ "$PARAM_AWS_CLI_DISABLE_PAGER" = 1 ]; then
        if [ -z "${AWS_PAGER+x}" ]; then
            echo 'export AWS_PAGER=""' >> "$BASH_ENV"
            echo "AWS_PAGER is being set to the empty string to disable all output paging for AWS CLI commands."
            echo "You can set the 'disable-aws-pager' parameter to 'false' to disable this behavior."
        fi
    fi
else
    echo "AWS CLI is already installed, skipping installation."
    aws --version
fi

AWS CLI is already installed, skipping installation.
aws-cli/1.16.209 Python/3.6.0 Windows/10 botocore/1.12.199
CircleCI received exit code 0
```

So as a result the windows executor fails at my next steps trying to perform docker login (as the get-login is now get-login-password etc.) to do a push to ECR.

**Comment by brivu** [comment]
This was resolved with PR #98
